Documentation/sourcebuild-in-repos/new-dependencies.md

```diff
@@ -0,0 +1,40 @@
+- If you are using the package in the actual build or want the version to be updated whenever the foreign repo publishes to your channel, use the version number property set up in eng/Versions.props.  This will be overridden in some cases by source-build.
+- If you are using an external or non-Arcade package, please coordinate as much as possible with other teams using that package.  Each package-version is essentially maintained as a separate concern, so something like repo A requiring Newtonsoft.Json 9.0.1 and repo B requiring 12.0.2 essentially doubles the source-build team's work.
+
+## Internal dependencies
+Adding a new dotnet dependency is usually pretty straightforward but does have some edge cases.  Use this checklist:
+1. Are you already using a package from the same repo and is the new dependency already source-built?
+	1. This is the simplest case.  You can add new dependencies from repos you are already using freely.  Note that you need to use the same version for all packages from the repo.
+2. Is the repo already built in source-build but the specific package is not?
+	1. There's probably an issue with source-building this package.  Please talk to a [source-build team member](https://github.com/orgs/dotnet/teams/source-build-internal) about why that is and whether we can fix it.
+2. Is this a Microsoft repo that uses Arcade to build?
+	1. Does the foreign repo depend on your repo, directly or indirectly?  i.e. would adding the dependency create a cycle?
+		1. This isn't necessarily a deal-breaker - it can sometimes be worked around with reference-only packages.  Please contact a [source-build team member](https://github.com/orgs/dotnet/teams/source-build-internal) to discuss.
+	2. Does the foreign repo publish to BAR?
+		1. If not, please contact them to get them publishing to BAR in an appropriate channel.
+	3. If neither of these caveats apply you should be in good shape.  Follow the instructions under "Basics" above.
+3. Source-build has in the past used Arcade shims to allow non-Arcade repos to build appropriate packages for source-build.  Please contact the source-build team to determine if this is a workable approach for your foreign repo.
+
+## External dependencies
+
+External (non-Microsoft or even non-dotnet) dependencies need to be carefully considered for source-build in addition to the licensing issues you are probably already familiar with.
+1. Microsoft but non-dotnet repos should usually be able to use the same Arcade wrapper to publish appropriate packages for source-build - please contact a source-build team member to discuss.
```

Comment:
I left the checklist combined because it's essentially the same for both of them but called out that .NET and non-.NET dependencies can be different.  This was redundant with the last step in the internal checklist so I removed it.